x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/analyzer/AnalyzerRules.java

```diff
@@ -138,14 +136,6 @@ public static List<Attribute> maybeResolveAgainstList(
             )
             .toList();
 
-        return singletonList(
-            ua.withUnresolvedMessage(
-                "Reference ["
-                    + ua.qualifiedName()
-                    + "] is ambiguous (to disambiguate use quotes or qualifiers); "
-                    + "matches any of "
-                    + refs
-            )
-        );
+        throw new IllegalStateException("Reference [" + ua.qualifiedName() + "] is ambiguous; " + "matches any of " + refs);
```

Comment:
Ambiguities were possible with `row` earlier; the message suggesting disambiguation does not make sense - that's not possible. This was carried over from `ql` and made sense for SQL. If we have ambiguities in ESQL, that's a bug IMO.